language/e2e-testsuite/src/tests/verify_txn.rs

```diff
@@ -241,35 +243,33 @@ fn verify_simple_payment() {
         StatusCode::GAS_UNIT_PRICE_ABOVE_MAX_BOUND
     );
 
+    // Test for a max_gas_amount that is insufficient to pay the minimum fee.
+    // Find the minimum transaction gas units and subtract 1.
+    let mut gas_limit = gas_constants
+        .to_external_units(gas_constants.min_transaction_gas_units)
+        .get();
+    if gas_limit > 0 {
+        gas_limit -= 1;
+    }
+    // Calculate how many extra bytes of transaction arguments to add to ensure
+    // that the minimum transaction gas gets rounded up when scaling to the
+    // external gas units. (Ignore the size of the script itself for simplicity.)
+    let mut extra_txn_bytes = 0;
+    if gas_constants.gas_unit_scaling_factor > gas_constants.min_transaction_gas_units.get() {
+        extra_txn_bytes = gas_constants.large_transaction_cutoff.get()
+            + (gas_constants.gas_unit_scaling_factor / gas_constants.intrinsic_gas_per_byte.get());
+    }
     let txn = sender
         .account()
         .transaction()
         .script(Script::new(
             p2p_script.clone(),
             vec![account_config::xus_tag()],
-            args.clone(),
-        ))
-        .sequence_number(10)
-        .max_gas_amount(1)
-        .gas_unit_price(GasConstants::default().max_price_per_gas_unit.get())
-        .sign();
-    assert_prologue_parity!(
-        executor.verify_transaction(txn.clone()).status(),
-        executor.execute_transaction(txn).status(),
-        StatusCode::MAX_GAS_UNITS_BELOW_MIN_TRANSACTION_GAS_UNITS
-    );
-
-    let txn = sender
-        .account()
-        .transaction()
-        .script(Script::new(
-            p2p_script.clone(),
-            vec![account_config::xus_tag()],
-            args.clone(),
+            vec![TransactionArgument::U8(42); extra_txn_bytes as usize],
```

Comment:
A bit puzzled here as well about the switch to U8 from the U64.

Reply:
Same as above -- this is just padding out the size of the transaction by `extra_txn_bytes`